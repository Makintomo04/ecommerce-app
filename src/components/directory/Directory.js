import React, { useState, useEffect, useRef } from "react";
import "./Directory.scss";
import Category from "../category/Category";
import { TweenMax, TimelineLite, Power3 } from "gsap";
const Directory = () => {
  const [categories, setCategories] = useState([
    {
      id: 1,
      title: "Mens",
      img_URL: `/Makintomo04/ecommerce-app/static/media/men3.a9dea04a.jpg`,
      link_URL: "mens",
    },
    {
      id: 2,
      title: "Womens",
      img_URL: "/Makintomo04/ecommerce-app/static/media/women2.69141f75.jpg",
      link_URL: "womens",
    },
    {
      id: 3,
      title: "Shoes",
      img_URL: "/Makintomo04/ecommerce-app/static/media/creps.2af0247c.jpg",
      link_URL: "shoes",
    },
    {
      id: 4,
      title: "Accessories",
      img_URL: "/Makintomo04/ecommerce-app/static/media/accessory.fac7c75c.jpg",
      link_URL: "accessories",
    },
    {
      id: 5,
      title: "Activewear",
      img_URL: "/Makintomo04/ecommerce-app/static/media/brand2.0a95648d.jpg",
      link_URL: "activewear",
    },
    {
      id: 6,
      title: "Clothing",
      img_URL: "/Makintomo04/ecommerce-app/static/media/nike.52712c27.jpg",
      link_URL: "clothing",
    },
  ]);
  let categoryRef = useRef(null);
  let tl = new TimelineLite();
  // useEffect(() => {
  //   tl.from(categoryRef, 2, { y: 200, ease: Power3.easeOut });
  // }, []);
  return (
    <div className="directory-menu">
      <div ref={(el) => (categoryRef = el)} className="directory-menu__item">
        {categories.map(({ id, ...remainingProps }) => (
          <Category key={id} {...remainingProps} />
        ))}
      </div>
    </div>
  );
};

export default Directory;
