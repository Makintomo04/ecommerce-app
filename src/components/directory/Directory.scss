.directory-menu {
  overflow: hidden;
  max-height: auto;
  margin-bottom: 80px;
  width: 100%;

  // height: 100%;
  justify-content: center;
  &__item {
    justify-content: center;
    width: 100%;
    display: grid;
    padding: 20px;
    margin-bottom: 50px;
    grid-template-areas:
      "men women brands shoes"
      "men  women accessories activewear ";

    grid-template-rows: 260px 260px;
    grid-template-columns: 300px 300px minmax(250px, auto) minmax(250px, auto);
    grid-gap: 17px;
  }
}

@media (max-width: 1440px) {
  .directory-menu__item {
    grid-template-areas:
      "men women shoes"
      "men  women accessories"
      "activewear activewear brands";
    grid-template-rows: 320px 320px 320px;
    grid-template-columns: minmax(200px, 400px) minmax(200px, 400px) minmax(
        200px,
        480px
      );
    grid-gap: 15px;
  }
}
@media (max-width: 1024px) {
  .directory-menu {
    margin: auto;
    &__item {
      grid-template-areas:
        "men women"
        "shoes accessories"
        "activewear brands";
      grid-template-rows: 520px 360px 360px;
      grid-template-columns: minmax(200px, 400px) minmax(200px, 400px);
      grid-gap: 15px;
    }
  }
}
@media (max-width: 600px) {
  .directory-menu {
    margin: auto;
    &__item {
      grid-template-areas:
        "men men"
        "women women"
        "shoes shoes"
        "accessories accessories"
        "activewear activewear"
        "brands brands";
      grid-template-rows: repeat(6, 280px);
      grid-template-columns: 1fr 1fr;
      grid-gap: 15px;
    }
  }
}
