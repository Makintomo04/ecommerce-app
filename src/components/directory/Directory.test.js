import React from "react"
import {shallow} from "enzyme"
import Directory from "./Directory"


it("Should render Directory component correctly", () => {
    const directory = shallow(<Directory />);
    expect.assertions(1)
    expect(directory.props().children).not.toBeUndefined() // check if a component renders correctly
  });