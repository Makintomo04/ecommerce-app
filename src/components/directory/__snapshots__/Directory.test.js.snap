// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Directory component correctly 1`] = `
<div
  className="directory-menu"
>
  <div
    className="directory-menu__item"
  >
    <withRouter(Category)
      img_URL="/Makintomo04/ecommerce-app/static/media/men3.a9dea04a.jpg"
      key="1"
      link_URL="mens"
      title="Mens"
    />
    <withRouter(Category)
      img_URL="/Makintomo04/ecommerce-app/static/media/women2.69141f75.jpg"
      key="2"
      link_URL="womens"
      title="Womens"
    />
    <withRouter(Category)
      img_URL="/Makintomo04/ecommerce-app/static/media/creps.2af0247c.jpg"
      key="3"
      link_URL="shoes"
      title="Shoes"
    />
    <withRouter(Category)
      img_URL="/Makintomo04/ecommerce-app/static/media/accessory.fac7c75c.jpg"
      key="4"
      link_URL="accessories"
      title="Accessories"
    />
    <withRouter(Category)
      img_URL="/Makintomo04/ecommerce-app/static/media/brand2.0a95648d.jpg"
      key="5"
      link_URL="activewear"
      title="Activewear"
    />
    <withRouter(Category)
      img_URL="/Makintomo04/ecommerce-app/static/media/nike.52712c27.jpg"
      key="6"
      link_URL="clothing"
      title="Clothing"
    />
  </div>
</div>
`;
